### DISCUSSION TITLE
Create fixture adhering to FluentValidation rules

### DISCUSSION THREAD

**Discussion by MichaelHochriegl** [AutoFixture/AutoFixture#1296] [Ideas]
Greetings all,

I really like AutoFixture, but I always struggle with using it for scenarios where I have validation based on [FluentValidation](https://github.com/FluentValidation/FluentValidation).
Right now AutoFixture supports only validation rules by DataAnnotations.

I poked around a couple of hours looking at a custom ISpecimenBuilder that get's the AbstractValidators (from FluentValidation) for the requested Type but I didn't really know I how to get further with this.

If there already is a way to get this working please let me know, couldn't find anything.
Otherwise I think this would be a nice addition to AutoFixture.

**Comment by Ergamon** [comment]
Sounds like a nice idea, but how to generate the data in non simplistic scenarios?

The DataAnnotations part of AutoFixture is not really working for multiple attributes.

I got some demo code working for this:
```
  public class CustomerValidator : AbstractValidator<Customer>
  {
    public CustomerValidator()
    {
      RuleFor(c => c.Id)
        .GreaterThan(42);

      RuleFor(c => c.Name)
        .MaximumLength(4);
    }
  }
```

So as long as you have one default validator for a property, my proof of concept code works.

But as soon as you have multiple validators and custom code it gets impossible to handle or how should the data be generated?

**Reply by aivascu** [reply]
@Ergamon multiple validators might still be fine, builders could be in most cases determined from the context. The tricky (impossible) part is indeed the custom validators. 

This might be a known limitation of the library, and the users will have to provide explicit builders for some members/types. This way the library would cover most use cases.

In any case I don't think this should be part of the official AutoFixture libraries. It is a pretty good idea for a community package though.

**Reply by MichaelHochriegl** [reply]
I'm up for it to be in a separate library that kinda glues AutoFixture and FluentValidation together.
Is there anything to look at at how to best get this separate library started?

@Ergamon do you have your demo code up in github somewhere? Would like to take a look at it if you don't mind.
